SpiralCompiler/Syntax/Lexer.cs

```diff
@@ -0,0 +1,217 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace SpiralCompiler.Syntax
+{
+    public class Lexer
+    {
+        private readonly string code;
+        private int index;
+
+        private bool IsEnd => index >= code.Length;
+
+        public Lexer(string code)
+        {
+            this.code = code;
+        }
+
+        public static List<Token> Lex(string code)
+        {
+            var lexer = new Lexer(code);
+            var tokenlist = new List<Token>();
+
+            while (true)
+            {
+                var token = lexer.Next();
+                tokenlist.Add(token);
+
+                if (token.Type == TokenType.EndOfFile) break;
+            }
+            return tokenlist;
+        }
+
+        public Token Next()
```

Comment:
If you have the previous method as public API, do you need this as public as well?
If no, you might even want to consider making the constructor and everything else private too.